src/tasks/DockerPseudoShell.ts

```diff
@@ -0,0 +1,109 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as cp from 'child_process';
+import { CancellationToken, CancellationTokenSource, Event, EventEmitter, Pseudoterminal, TerminalDimensions, WorkspaceFolder } from 'vscode';
+import { CommandLineBuilder } from '../utils/commandLineBuilder';
+import { resolveVariables } from '../utils/resolveVariables';
+import { DockerBuildTask } from './DockerBuildTaskProvider';
+import { DockerRunTask } from './DockerRunTaskProvider';
+import { DockerTaskProviderBase } from './DockerTaskProviderBase';
+import { DockerTaskExecutionContext } from './TaskHelper';
+
+const DEFAULT = '0m';
+const DEFAULTBOLD = '0;1m';
+const RED = '31m';
+const YELLOW = '33m';
+
+export class DockerPseudoShell implements Pseudoterminal {
+    private readonly closeEmitter: EventEmitter<number> = new EventEmitter<number>();
+    private readonly writeEmitter: EventEmitter<string> = new EventEmitter<string>();
+    private readonly cts: CancellationTokenSource = new CancellationTokenSource();
+
+    public onDidWrite: Event<string> = this.writeEmitter.event;
+    public onDidClose: Event<number> = this.closeEmitter.event;
+
+    constructor(private readonly taskProvider: DockerTaskProviderBase, private readonly task: DockerBuildTask | DockerRunTask) { }

```

Comment:
Nit: if other types refer to the base class `DockerTaskProviderBase`, it shouldn't end in the `Base` suffix.  Maybe just use `DockerTaskProvider`?